src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py

```diff
@@ -272,6 +275,43 @@ def getDatasetLocations(self, datasets, account=None):
                                                              account=account)
             result[datasetPath] = self.cric.PNNstoPSNs(locations)
         return result
+    
+    def getDatasetLocationsFromMSPileup(self, datasetsWithDbsURL):
+        """
+        Returns a dictionary with the location of the datasets according to MSPileup
+        The definition of "location" here is a union of all sites holding at least
+        part of the dataset (defined by the DATASET grouping).
+        :param datasets: list of datasets
+        :param account: rucio account to use, if it is not provided we fallback to
+        pileup account for backward compatibility
+        :return: a dictionary of dataset locations, key'ed by the dataset name
+        """
+        
+        result = {}
+        for dbsUrl, datasets in datasetsWithDbsURL.items():
+            for dataset in datasets:
+                try:
+                    queryDict = {'query': {'pileupName': dataset},
+                                'filters': ['expectedRSEs', 'currentRSEs', 'pileupName', 'containerFraction', 'ruleIds']}
+                    pileUpinstance = '-testbed' if 'cmsweb-testbed' in dbsUrl else '-prod'
```

Comment:
These 2 lines with the url logic don't need to be executed for each dataset. So they can move under the first `for` loop instead of the inner one.

Reply:
Ok, this would imply replacing `dataset` with `datasets` in [these lines](https://github.com/dmwm/WMCore/blob/e3798208ccd82db54431967d2cdf67899f26cb6a/src/python/WMCore/WorkQueue/Policy/Start/StartPolicyInterface.py#L306-L309). I am wondering if this further modification would work or not. My original idea relates to the implementation made by @d-ylee [here](https://github.com/d-ylee/WMCore/blob/84de26e03211212a6c365ef93c3f5081e2937986/src/python/WMCore/WorkQueue/DataLocationMapper.py#L143-L157) of the `MSPileupUtils`

Reply:
My suggestion is to implement something like:
```
for dbsUrl, datasets in datasetsWithDbsURL.items():
    pileUpinstance = '-testbed' if 'cmsweb-testbed' in dbsUrl else '-prod'
    msPileupUrl = f"https://cmsweb{pileUpinstance}.cern.ch/ms-pileup/data/pileup"
    self.logger.info(f'Will fetch {len(datasets)} from MSPileup url: {dbsUrl}')
    for dataset in datasets:
        queryDict = {'query': {'pileupName': dataset},
            'filters': ['expectedRSEs', 'currentRSEs', 'pileupName', 'containerFraction', 'ruleIds']}
        try:
            doc = getPileupDocs(msPileupUrl, queryDict, method='POST')[0]
 ....
```
does it make sense to you?